File: php/98-validate-binary-search-tree.php
<?php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($value) { $this->val = $value; }
 * }
 */
class Solution 
{
    private $data = [];
    /**
     * @param TreeNode $root
     * @return Boolean
     */
    public function isValidBST($root) 
	{
        $this->inorder($root);
        $len = count($this->data);
        for ($i = 1; $i < $len; $i++) {
            $prev = $this->data[$i-1];
            if ($prev >= $this->data[$i]) {
                return false;
            }
        }
        return true;
    }
    private function inorder($node) 
	{
        if ($node == null) {
            return ;
        }
        $this->inorder($node->left);
        $this->data[] = $node->val;
        $this->inorder($node->right);
    }
}

File: go/98-validate-binary-search-tree.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func isValidBST(root *TreeNode) bool {
	data := []int{}
	inorder(root, &data)
	for i := 1; i < len(data); i++ {
		if data[i-1] >= data[i] {
			return false
		}
	}
	return true
}
func inorder(root *TreeNode, data *[]int) {
	if root == nil {
		return
	}
	inorder(root.Left, data)
	*data = append(*data, root.Val)
	inorder(root.Right, data)
}
